Gems/Atom/RHI/Vulkan/Code/Source/RHI/FrameGraphExecuter.cpp

```diff
@@ -55,11 +59,24 @@ namespace AZ
         void FrameGraphExecuter::BeginInternal(const RHI::FrameGraph& frameGraph)
         {
             Device& device = GetDevice();
-
+            AZStd::vector<const Scope*> mergedScopes;
+#if defined(AZ_FORCE_CPU_GPU_INSYNC)
+            // Forces all scopes to issue a dedicated merged scope group with one command list.
+            // This will ensure that the Execute is done on only one scope and if an error happens
+            // we can be sure about the work gpu was working on before the crash.
+            for (const RHI::Scope* scopeBase : frameGraph.GetScopes())
+            {
+                const Scope& scope = *static_cast<const Scope*>(scopeBase);
+                mergedScopes.push_back(&scope);
```

Comment:
Same as for metal, seems unnecessary to deref a ptr you are immediately going to get the address of, and that is unused afterwards.